### DISCUSSION TITLE
How to display the progress bar for parallel processes using rstudioapi::jobRunScript?

### DISCUSSION THREAD

**Discussion by djfgerber** [futureverse/progressr#141] [Q&A]
This question was already posted on stackexchange with no answers. Maybe the authors of this package know more about this or can give a lead on how to circumvent the problem or where to find answers?

I have a `R` script called `parallel_progress_reprex.R` that executes code in parallel and tracks the progress using `progressr`:

```
library(progressr) ## use progressr for procession updates
library(doFuture)  ## attaches also foreach and future
registerDoFuture() ## tell foreach to use futures
plan(multisession) ## parallelize over a local PSOCK cluster

xs <- 1:8

with_progress({
  p <- progressor(along = xs) 
  y <- foreach(x = xs) %dopar% {
    p()                       
    Sys.sleep(1)
    sqrt(x)
  }
})
```
When execuded in the interactive mode or with `source("parallel_progress_reprex.R")` this displays a progress bar which tracks my progress. However, when I use the `rstudioapi::jobRunScript`` command the progress bar is not displayed anymore.
```
rstudioapi::jobRunScript("parallel_progress_reprex.R")
```
How can the progress bar be forced to show up using `rstudioapi::jobRunScript`?

**Comment by scottkosty** [comment]
@djfgerber Thanks for mentioning your other question. Indeed, it's helpful to cross-link them. That way, if an answer is posted here but someone has the same question and their search takes them to the other thread, they can still find the answer. I think I found your question on Stack Overflow [here](https://stackoverflow.com/questions/72962768/how-to-display-the-progress-bar-for-parallel-processes-using-rstudioapijobruns).

**Reply by djfgerber** [reply]
Thanks for the suggestion, this is exactly the question on Stackexchange, I forgot to link them in my post.
